mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -2889,6 +2970,125 @@ pub fn tendermint_priv_key_policy(
     }
 }
 
+pub(crate) fn chain_registry_name_from_account_prefix(ctx: &MmArc, prefix: &str) -> Option<String> {
+    let Some(coins) = ctx.conf["coins"].as_array() else {
+        return None;
+    };
+
+    for coin in coins {
+        let protocol = coin
+            .get("protocol")
+            .unwrap_or(&serde_json::Value::Null)
+            .get("type")
+            .unwrap_or(&serde_json::Value::Null)
+            .as_str();
+
+        if protocol != Some(TENDERMINT_COIN_PROTOCOL_TYPE) {
+            continue;
+        }
+
+        let coin_account_prefix = coin
+            .get("protocol")
+            .unwrap_or(&serde_json::Value::Null)
+            .get("protocol_data")
+            .unwrap_or(&serde_json::Value::Null)
+            .get("account_prefix")
+            .map(|t| t.as_str().unwrap_or_default());
+
+        if coin_account_prefix == Some(prefix) {
+            return coin
+                .get("protocol")
+                .unwrap_or(&serde_json::Value::Null)
+                .get("protocol_data")
+                .unwrap_or(&serde_json::Value::Null)
+                .get("chain_registry_name")
+                .map(|t| t.as_str().unwrap_or_default().to_owned());
+        }
+    }
+
+    None
+}
+
+pub async fn get_ibc_transfer_channels(
+    source_registry_name: String,
+    destination_registry_name: String,
+) -> IBCTransferChannelsResult {
+    #[derive(Deserialize)]
+    struct ChainRegistry {
+        channels: Vec<IbcChannel>,
+    }
+
+    #[derive(Deserialize)]
+    struct ChannelInfo {
+        channel_id: String,
+        port_id: String,
+    }
+
+    #[derive(Deserialize)]
+    struct IbcChannel {
+        #[allow(dead_code)]
+        chain_1: ChannelInfo,
+        chain_2: ChannelInfo,
+        ordering: String,
+        version: String,
+        tags: Option<IBCTransferChannelTag>,
+    }
+
+    let source_filename = format!("{}-{}.json", source_registry_name, destination_registry_name);
+    let git_controller: GitController<GithubClient> = GitController::new(GITHUB_API_URI);
+
+    let metadata_list = git_controller
+        .client
+        .get_file_metadata_list(
+            CHAIN_REGISTRY_REPO_OWNER,
+            CHAIN_REGISTRY_REPO_NAME,
+            CHAIN_REGISTRY_BRANCH,
+            CHAIN_REGISTRY_IBC_DIR_NAME,
+        )
+        .await
+        .map_err(|e| IBCTransferChannelsRequestError::Transport(format!("{:?}", e)))?;
```

Comment:
I know it's our repo, but is it completely secure to rely on this repo for ibc channels data? We have to consider supply chain attacks, what would happen if a wrong channel id is used for a transfer?
P.S. the alternative would be to load a file like the coins file and it will be updated with each new GUI release while relying on the user to enter custom channel ids for new IBC channels similar to custom token imports. What do you think @onur-ozkan?

c.c. @ca333

Reply:
> I know it's our repo, but is it completely secure to rely on this repo for ibc channels data? We have to consider supply chain attacks, what would happen if a wrong channel id is used for a transfer?

Either our repo or user's hostname configuration must be hacked (which is already ..... situation) in order to provide incorrect channels. But that's already more dangerous than providing incorrect channels.

The things which may happen when using wrong channel:

- By default, if using a channel id that doesn't exists, funds will be recovered (except the tx fee part).

- If channel id belongs to a relayer that is customized by malicious person, then anything can happen depending on what this person changed on his/her relayer.  

> P.S. the alternative would be to load a file like the coins file and it will be updated with each new GUI release while relying on the user to enter custom channel ids for new IBC channels similar to custom token imports. What do you think @onur-ozkan?

Channels can be changed very frequently, I don't think we can keep up with it by releasing mm2 for each change on the channels.

Reply:
> Either our repo or user's hostname configuration must be hacked (which is already ..... situation) in order to provide incorrect channels. But that's already more dangerous than providing incorrect channels.

Only one dev with write auth getting hacked is enough to write incorrect channels info to this repo and it will be used straight away by komodo wallet and the defi framework release. We can't lock this to a certain commit hash too.

> Channels can be changed very frequently, I don't think we can keep up with it by releasing komodo-wallet for each change on the channels.

We can leave it as is for now but I wanted to raise this issue for the sec review team to see their opinion.
One more question, do channels change for already running and established chains or is this needed for new chains to be always available in komodo wallet?

Reply:
> Only one dev with write auth getting hacked is enough to write incorrect channels info to this repo and it will be used straight away by komodo wallet and the defi framework release. We can't lock this to a certain commit hash too.

We can but that kills the whole idea. I would just put the json file in the mm2 source code instead of using the locked hash, but as I mentioned above we can not make mm2 release each time a channel info changes in the repo. Giving the write access to someone irrelevant is not an easy accident and that's already pretty bad situation in my opinion.

> We can leave it as is for now but I wanted to raise this issue for the sec review team to see their opinion.
One more question, do channels change for already running and established chains or is this needed for new chains to be always available in komodo wallet?

They can be removed/updated for any new/old chain.